File: testone.ts
const enum testoneChoice {
    one,
    two
}

namespace testone {

    /**
    * Display one.
    */
    //% shim=testone::one
    export function one(): void { basic.showString("one") }

    /**
    * Display two.
    */
    //% shim=testone::two
    export function two(): void { basic.showString("two") }

    /**
    * Display any.
    */
    //% shim=testone::any
    export function any(choice: testoneChoice): void { basic.showString("any") }

}

File: testone.cpp
#include "pxt.h"

using namespace pxt;

enum class testoneChoice {
    one,
    two
};

namespace testone {

//%
void one() {
    uBit.display.scroll("11");
}

//%
void two() {
    uBit.display.scroll("22");
}

//%
void any( testoneChoice choice) {
  switch (choice)
  {
    case testoneChoice::one:
      testone::one();
      break;
    case testoneChoice::two:
      testone::two();
      break;
  }
}

} // namespace testone
